assignment2/q2.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+using namespace std;
+
+struct Node{
+    int value;
+    Node *parent;
+    Node *left, *right;
+};
+
+bool FindNode(Node *root, Node *to_find) {
+
+    if(root == NULL)
+        return false;
+
+    if(root == to_find)
+        return true;
+
+    if(FindNode(root->left, to_find) || FindNode(root->right, to_find))
+        return true;
+
+    return false;
+}
+
+//O(N^2) without any additional memory or changing structure
+//could be O(N) if i.e. it stores paths to nodes and then compares them
+Node *FindCommonAncestor(Node *root, Node *first, Node *second) {
```

Comment:
Nice solution, straightforward and readable. And well commented too! :+1: 
As you wrote, you could improve the complexity in various ways.